src/lib/mlang.js

```diff
@@ -147,108 +259,207 @@ foam.CLASS({
 });
 
 foam.CLASS({
-  package: 'foam.mlang',
+  package: 'foam.mlang.predicate',
+  name: 'AndExpr',
+  extends: 'foam.mlang.predicate.NaryExpr',
+  methods: [
+    function f(o) {
+      for ( var i = 0 ; i < this.args.length ; i++ ) {
+        if ( ! this.args[i].f(o) ) return false;
+      }
+      return true;
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.mlang.predicate',
   name: 'ContainsExpr',
-  extends: 'foam.mlang.BINARY',
+  extends: 'foam.mlang.predicate.BinaryExpr',
   methods: [
     function f(o) {
-      if ( this.arg1.f(o).indexOf(this.arg2.f(o)) != -1 ) return true;
-      return false;
+      var s1 = this.arg1.f(o);
+      return s1 ? s1.indexOf(this.arg2.f(o)) !== -1 : false;
     }
   ]
 });
 
 foam.CLASS({
-  package: 'foam.mlang',
+  package: 'foam.mlang.predicate',
+  name: 'ContainsICExpr',
+  extends: 'foam.mlang.predicate.BinaryExpr',
+  methods: [
+    function f(o) {
+      var s1 = this.arg1.f(o).toUpperCase();
+      var s2 = this.arg2.f(o).toUpperCase();
+      return s1 ? s1.indexOf(s2) !== -1 : false;
```

Comment:
If you going to guard against s1 being undefined you need to do that before you call toUpperCase()